common/mb_odometry.c

```diff
@@ -32,6 +33,17 @@ void mb_initialize_odometry(mb_odometry_t* mb_odometry, float x, float y, float
 *
 *******************************************************************************/
 void mb_update_odometry(mb_odometry_t* mb_odometry, mb_state_t* mb_state){
+    float diff = mb_state->yaw_delta - mb_state->gyro_heading_delta;
+    
+    if (abs(diff) > 0.1 && (mb_state->left_velocity != 0 || mb_state->right_velocity != 0)) {
```

Comment:
the threshold 0.1 here is a user-defined constant to check if it is necessary to switch to gyro heading, might need to use #define DELTA_THETA_THRESHOLD 0.1

Reply:
Great idea, added it to mb_defs.